internal/trie/node/variants.go

```diff
@@ -23,4 +23,28 @@ var (
 		bits: 0b1100_0000,
 		mask: 0b1100_0000,
 	}
+	leafWithHashedValueVariant = variant{ // leaf containing hashes 001
+		bits: 0b0010_0000,
+		mask: 0b1110_0000,
+	}
+	branchWithHashedValueVariant = variant{ // branch containing hashes 0001
+		bits: 0b0001_0000,
+		mask: 0b1111_0000,
+	}
+	emptyVariant = variant{ // empty 0000 0000
+		bits: 0b0000_0000,
+		mask: 0b1111_1111,
+	}
+	compactEncodingVariant = variant{ // compact encoding 0001 0000
+		bits: 0b0000_0001,
+		mask: 0b1111_1111,
+	}
 )
+
+// partialKeyLengthHeaderMask returns the partial key length
+// header bit mask corresponding to the variant header bit mask.
+// For example for the leaf variant with variant mask 1100_0000,
+// the partial key length header mask returned is 0011_1111.
+func (v variant) partialKeyLengthHeaderMask() byte {
```

Comment:
Just to make sure I understand, the partial key length header mask is supposed to be the logical negation of the variant header bit mask?

Reply:
Jimmy, it's explained with more [detail here](https://spec.polkadot.network/chap-state#defn-node-header)

It is as you said, the mask is the logical negation of the variant header bit mask.
For example if you have a leaf variant that is represented by 01 (0100_0000), the mask will be 1100_0000 so the partial key length mask will be 6 bits long 0011_1111 that is the same that ^1100_000.

But I think it's more clear in the spec section I sent you above